### DISCUSSION TITLE
Xcode version support

### DISCUSSION THREAD

**Discussion by brentleyjones** [bazelbuild/rules_apple#1914] [General]
Next month Apple won’t allow you to submit to the App Store using anything before Xcode 14.1. We (the rules_apple/rules_swift maintainers) have used this as the highest our minimum Xcode version support could be. But what about the absolute minimum we should support? There are use cases outside of App Store submission, sure, but do you also need the latest rules version in those cases as well?

I’m also personally interested from the point of view of rules_xcodeproj.

So, what is your thoughts on this? Which Xcode version should we minimally support/which methodology should we use to determine it?

**Comment by luispadron** [comment]
My very simple take is to make minimum supported by the _current_ ruleset whatever the minimum supported Xcode version is. Like you mentioned folks who need older versions could always use the older rules. Might be helpful to describe what would be required maintenance wise from contributors to support older versions in the current rules.

If it's much more work I think it makes sense to drop older version

**Comment by acecilia** [comment]
> Which Xcode version should we minimally support/which methodology should we use to determine it?

For official support (with CI test/validations), I would say your suggestion of matching AppStore makes sense. If somebody needs an older version of Xcode, considering this would be the lesser of the cases, I would say is easier if they try to update (easier for you maintainers, and probably beneficial for users in the long term, as support/documentation for older versions will be harder due to almost nobody using such version of Xcode).

Of course, if no breaking changes, unofficially rules could keep working with older versions of Xcode. But I would say the burden of officially maintaining those (with tests) is probably not worth it
